presto-main/src/main/java/io/prestosql/dispatcher/DispatchManager.java

```diff
@@ -261,6 +263,22 @@ private boolean queryCreated(DispatchQuery dispatchQuery)
                 .collect(toImmutableList());
     }
 
+    @Managed
+    public long getQueuedQueries()
+    {
+        return queryTracker.getAllQueries().stream()
```

Comment:
Will this scale for cluster with large number of concurrent queries? The JMX scrape intervals can be as low as 10 seconds. So is calling `getAllQueries()` at 6 rpm OK for the cluster?

I too prefer this change since everything is simpler now but am concerned about the above point.

Reply:
It's not an expensive operation.  The UI actually calls this (see https://github.com/prestosql/presto/blob/master/presto-main/src/main/java/io/prestosql/server/ui/ClusterStatsResource.java#L78) more often than that.